quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakMain.java

```diff
@@ -164,17 +170,41 @@ public int run(String... args) throws Exception {
     private void createAdminUser() {
         String adminUserName = System.getenv(KEYCLOAK_ADMIN_ENV_VAR);
         String adminPassword = System.getenv(KEYCLOAK_ADMIN_PASSWORD_ENV_VAR);
+        String tmpFirstName = System.getenv(KEYCLOAK_ADMIN_FIRSTNAME_ENV_VAR);
+        String tmpLastName = System.getenv(KEYCLOAK_ADMIN_LASTNAME_ENV_VAR);
+        String tmpEmail = System.getenv(KEYCLOAK_ADMIN_EMAIL_ENV_VAR);
 
-        if ((adminUserName == null || adminUserName.trim().length() == 0)
-                || (adminPassword == null || adminPassword.trim().length() == 0)) {
+        if (StringUtil.isBlank(adminUserName) || StringUtil.isBlank(adminPassword)) {
             return;
         }
 
+        // try to create admin user only with username and password
+        if (StringUtil.isBlank(tmpFirstName)) {
+            tmpFirstName = adminUserName;
+        }
+
+        if (StringUtil.isBlank(tmpLastName)) {
+            tmpLastName = adminUserName;
+        }
+
+        if (StringUtil.isBlank(tmpEmail)) {
+            tmpEmail = adminUserName + "@" + KEYCLOAK_ADMIN_DEFAULT_EMAIL_DOMAIN;
+        }
+
+        if (!EmailValidationUtil.isValidEmail(tmpEmail)) {
+            return;
```

Comment:
As mentioned above, I would add something to the server.log for this case to indicate that admin user was not created and the reason of this (possibly also with a hint to explicitly set `KEYCLOAK_ADMIN_EMAIL` with a valid email).

Reply:
Sounds good, Thanks